### PR TITLE
Better documentation for Cube names in curl

### LINKED ISSUE #3563: analysis.calculus directional_names documentation
`calculus.curl` expects input cubes to begin with their directional name followed by whitespace/underscore - achieved via `calculus.spatial_vectors_with_phenom_name`. This behaviour should be documented.

The list of accepted directional names should also be documented. This is currently held in the `directional_names` set within `spatial_vectors_with_phenom_name`, which is a private method. To expose the list for documentation `directional_names` needs to become global, and to get Sphinx to pick up the values and any lines of text I think we would need to use [Sphinx autodata](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autodata). May need some sphinx config changes but not sure yet.

This is something I will need to learn more about when I have some time, so just documenting in an issue for now. Any better ideas for how to document `directional_names` will be welcomed.

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Closes #3563

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: lib/iris/analysis/calculus.py [hunk 1]
```diff
@@ -471,6 +471,9 @@ def curl(i_cube, j_cube, k_cube=None):
     Calculate the 2-dimensional or 3-dimensional spherical or cartesian
     curl of the given vector of cubes.
 
+    The cube standard names must match one the combinations in
```
Comment (ESadek-MO) [inline lib/iris/analysis/calculus.py:474]: one of the

File: lib/iris/analysis/calculus.py [hunk 2]
```diff
@@ -734,12 +737,27 @@ def curl(i_cube, j_cube, k_cube=None):
...
+)
+
+
 def spatial_vectors_with_phenom_name(i_cube, j_cube, k_cube=None):
     """Given spatially dependent cubes, return a list of the spatial coordinate names.
 
     Given 2 or 3 spatially dependent cubes, return a list of the spatial
     coordinate names with appropriate phenomenon name.
 
+    The cube standard names must match one the combinations in
```
[the remark this answers is not in the document]
Comment (trexfeathers) [reply lib/iris/analysis/calculus.py]: 159e53c6b59c1eb8e7506dbe6bd01764665baa8f

File: lib/iris/analysis/calculus.py (hunk 1 above)
Comment (trexfeathers) [reply lib/iris/analysis/calculus.py]: 159e53c6b59c1eb8e7506dbe6bd01764665baa8f

**Review by ESadek-MO — APPROVED** [review]
Awesome, looks good to me, thanks @trexfeathers!

**Pull request merged**